src/vm.c

```diff
@@ -159,71 +159,54 @@ static val_t v7_create_object(struct v7 *v7, struct v7_object *proto) {
   }
   (void) v7;
   o->properties = NULL;
-  o->prototype = proto;
+  o->prototype = prototype == V7_UNDEFINED ? NULL : val_to_pointer(prototype);
```

Comment:
let's use `V7_NULL` here for `NULL`, since that's the value that will be passed by the `Obj_create` primitive (which accepts JS `null` as a valid value, while it rejects `undefined`. V8 says:

```
TypeError: Object prototype may only be an Object or null
```

Spidermonkey:
```
TypeError: Expected argument of type object, but instead had type undefined
```